### DISCUSSION TITLE
Squash all commits above selected

### DISCUSSION THREAD

**Discussion by okuuva** [jesseduffield/lazygit#2440] [Q&A]
I know there's a way to squash all `!fixup` commits in `lazygit` above selected commit but is there a way to squash ***all*** commits above selected commit?

Somewhat related to #1160

**Comment by stefanhaller** [comment]
Right now the best you can do is to hit "e" on the selected commit, and then keep pressing up-arrow and "f" to change all the "pick" lines into "fixup", until you are at the top of the branch. Then hit "m", enter to continue the rebase. Kind of painful if you have hundreds of commits to squash, but hopefully your branches aren't _that_ long. 😄

**Comment by danielo515** [comment]
Is still no way of doing this? I want to locate all the commits I want to merge into one, hit the keymap and get a prompt for the final commit message. Is this not possible?

**Comment by AndrewSav** [comment]
I think [this issue](https://github.com/jesseduffield/lazygit/issues/2249) if resolved will also cover this use case

**Comment by danielo515** [comment]
Yes, it looks like that

On Wed, May 17, 2023 at 8:47 AM Andrew Savinykh ***@***.***>
wrote:

> I think this issue <https://github.com/jesseduffield/lazygit/issues/2249>
> if resolved will also cover this use case
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/jesseduffield/lazygit/discussions/2440#discussioncomment-5923984>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AARKJWIK7N4Q4JQFPP2OJC3XGRX7JANCNFSM6AAAAAAUZKB5Z4>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
---
https://danielorodriguez.com
